### PR TITLE
Tel 82 create post source endpoint

### PR DESCRIPTION
**Pull request by NagarajNidagundi**

### CODE REVIEW

File: src/WWTelematics.SourceService/Application/SourceManager.cs [hunk 1]
```diff
@@ -123,5 +124,55 @@ public ApplicationLoadBalancerResponse GetSourceList(IDictionary<string, string>
...
+            Source source;
+
+            if (string.IsNullOrEmpty(sourceRequest))
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(400);
+            }
+
+            try
+            {
+                source = JsonConvert.DeserializeObject<Source>(sourceRequest);
```
Comment (AmitkumarEvry) [inline src/WWTelematics.SourceService/Application/SourceManager.cs]: Post input/output json should be in below format.

{

	“ClientGroup”: {

		“id”: “11111111-2222-3333-4444-555555555555”,

		“label”: “Label 1”,

		“created_time”: “2020-02-01T00:00:00”

	}

}

File: src/WWTelematics.SourceService/Validators/SourceValidator.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public class SourceValidator : AbstractValidator<Source>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SourceValidator"/> class.
+        /// </summary>
+        public SourceValidator()
+        {
+            this.RuleFor(x => x.Label).NotEmpty().NotNull().Length(1, 31)
+                     .WithMessage("Label text length should be between 1 to 31.")
+                .Must(m => !m.Contains("|", StringComparison.CurrentCultureIgnoreCase))
```
Comment (AmitkumarEvry) [inline src/WWTelematics.SourceService/Validators/SourceValidator.cs]: This will throw exception if label is not passed  in json or label is null/empty.

**Comment by sinaj-evry** [comment]
fix amit comments otherwise its looks good

File: src/WWTelematics.SourceService/Validators/SourceValidator.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    /// Source model Validator.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1710:Identifiers should have correct suffix", Justification = "Validators implement IEnumerable to allow looking at rules, but are not fundamentally collections.")]
+
+    public class SourceValidator : AbstractValidator<Source>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SourceValidator"/> class.
+        /// </summary>
+        public SourceValidator()
```
Comment (davidw-foxtrax) [inline src/WWTelematics.SourceService/Validators/SourceValidator.cs]: Please validate the remaining fields.

For example, POSTing
```
"body": "{	\"source\": {		\"label\": \"Posted Source\"	}}"
```

Throws a database exception
```
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.

[...]

Npgsql.PostgresException: 23502: null value in column "tenant_source" violates not-null constraint
```

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: 7fb76121493254eea94358e7bebdde20f92731db
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fc3dc5ee-c07f-4d04-ac73-5c281fd4a930%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.SourceService/Validators/SourceValidator.cs (hunk 3 above)
Comment (NagarajNidagundi) [reply src/WWTelematics.SourceService/Validators/SourceValidator.cs]: done - validated the remaining fields.

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: ee9879df899b78ef55ca49adc7ea3ad7b59148ea
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fcc91447-be04-4cd7-9a5b-42dc4f569b69%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: e83f22974c771ac6ce0841c6fa961ab82b025f19
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=df11dd69-4117-4ba5-8a98-1443c9b95155%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: 40399e7088acf72209041c9e1fabddc88b571267
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ef7b3df6-c54b-47d3-bd8d-ebe7fed80a50%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by NagarajNidagundi** [comment]
@davidw-foxtrax - We can't directly accept the nested JSON elements as JSON instead in strings.

**Comment by davidw-foxtrax** [comment]
Approving PR as is.  Going forward, we will need to investigate how to accept JSON from clients as this is what they will expect to send.

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: 57274e91bd85a432e40217eb87159471494c71c2
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=81abcb31-442c-4f05-822f-1e450d6b7823%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**